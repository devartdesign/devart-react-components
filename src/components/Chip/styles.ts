import { StyleSheet } from 'aphrodite/no-important';
import { Color } from 'devart-utils/dist/models/general';

export default StyleSheet.create({
  container: {
    cursor: 'pointer',
    color: 'rgba(0, 0, 0, 0.87)',
    border: 'none',
    height: '32px',
    display: 'inline-flex',
    outline: '0',
    padding: '0',
    fontSize: '0.8125rem',
    boxSizing: 'border-box',
    transition: 'background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
    alignItems: 'center',
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    whiteSpace: 'nowrap',
    borderRadius: '16px',
    verticalAlign: 'middle',
    justifyContent: 'center',
    textDecoration: 'none',
    backgroundColor: '#e0e0e0',
  },
  label: {
    cursor: 'inherit',
    display: 'flex',
    alignItems: 'center',
    userSelect: 'none',
    whiteSpace: 'nowrap',
    paddingLeft: 12,
    paddingRight: 12,
  },
  labelWithIcon: {
    paddingRight: 0
  },
  icon: {
    fill: Color.PRIMARY
  }
});
