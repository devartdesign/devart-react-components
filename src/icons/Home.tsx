import React, { memo } from 'react';

interface IHomeIconProps {
  width?: number;
  height?: number;
  className?: string;
}

const HomeIcon = ({ width = 24, height = 24, className }: IHomeIconProps) => {
  return (
    <svg key="HomeIcon" className={className} xmlns="http://www.w3.org/2000/svg" width={width} height={height} viewBox="0 0 24 24">
      <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"/>
      <path d="M0 0h24v24H0z" fill="none"/>
    </svg>
  );
};

export default memo(HomeIcon);
