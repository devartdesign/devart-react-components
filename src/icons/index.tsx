export { default as SearchIcon } from './Search';
export { default as HomeIcon } from './Home';
export { default as DoneIcon } from './Done';
export { default as HamburgerIcon } from './Hamburger';
