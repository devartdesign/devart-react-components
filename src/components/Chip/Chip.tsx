import React, { memo } from 'react';
import { css } from 'aphrodite/no-important';

import styles from './styles';

export interface IChipProps {
  label: string;
  icon: any;
  key?: string | number;
  className?: string;
  onClick: () => void;
}

const Chip = ({ onClick, label, key, icon, className }: IChipProps) => {
  const Icon = icon;
  return (
    <div role="button" key={key} className={`${css(styles.container)} ${className}`} onClick={onClick}>
      <span className={css(styles.label)}>{label}</span>
      {icon && <Icon />}
    </div>
  );
};

export default memo(Chip);
