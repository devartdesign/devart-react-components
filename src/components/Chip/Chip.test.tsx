import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import { DoneIcon } from '../../icons';
import Chip, { IChipProps } from './Chip';

describe('Chip Component', () => {
  let Component;
  let props: IChipProps;

  beforeEach(() => {
    props = {
      label: 'label chip',
      icon: DoneIcon,
      key: 'key chip',
      onClick: jest.fn(),
    };
    Component = shallow(<Chip {...props} />);
  });

  it('should render', () => {
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should call onClick callback from props', () => {
    Component.find('div').simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });
});
