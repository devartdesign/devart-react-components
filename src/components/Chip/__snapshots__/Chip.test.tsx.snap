// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip Component should render 1`] = `
.container {
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  cursor: pointer;
  color: rgba(0, 0, 0, 0.87);
  border: none;
  height: 32px;
  display: -webkit-inline-box;
  display: -moz-inline-box;
  display: -ms-inline-flexbox;
  display: -webkit-inline-flex;
  display: inline-flex;
  outline: 0;
  padding: 0;
  font-size: 0.8125rem;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  -moz-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  -webkit-align-items: center;
  align-items: center;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  white-space: nowrap;
  border-radius: 16px;
  vertical-align: middle;
  -webkit-justify-content: center;
  justify-content: center;
  text-decoration: none;
  background-color: #e0e0e0;
  margin-right: 5px;
}

.label-labelWithIcon {
  -webkit-box-align: center;
  -ms-flex-align: center;
  cursor: inherit;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-align-items: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  padding-left: 12px;
  padding-right: 0px;
}

.icon {
  fill: #29434e;
  margin: 0 10px 0 5px;
}

<div
  className="container undefined"
  onClick={[MockFunction]}
  role="button"
>
  <span
    className="label-labelWithIcon"
  >
    label chip
  </span>
  <svg
    className="icon"
    height={18}
    viewBox="0 0 24 24"
    width={18}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
    />
  </svg>
</div>
`;
